edgelet/edgelet-utils/src/ser_de.rs

```diff
@@ -66,14 +67,75 @@ where
         .context(ErrorKind::SerdeClone)?)
 }
 
+pub fn deserialize_map_with_default_values<'de, K, V, D>(
+    deserializer: D,
+) -> StdResult<BTreeMap<K, V>, D::Error>
+where
+    K: Deserialize<'de> + Eq + std::hash::Hash + std::cmp::Ord,
+    V: Deserialize<'de> + Default,
+    D: Deserializer<'de>,
+{
+    // Loosely derived from https://serde.rs/deserialize-map.html
+    // Create a MapVisitor which will read a map as HashMap<K,Option<V>
+    // In the case of a map we need generic type parameters K and V to be
+    // able to set the output type correctly, but don't require any state.
+    // This is an example of a "zero sized type" in Rust. The PhantomData
+    // keeps the compiler from complaining about unused generic type
+    // parameters.
+    struct OptionalValueVisitor<K, V> {
+        marker: PhantomData<fn() -> (K, V)>,
+    };
+
+    impl<K, V> OptionalValueVisitor<K, V> {
+        fn new() -> Self {
+            OptionalValueVisitor {
+                marker: PhantomData,
+            }
+        }
+    }
+
+    impl<'de, K, V> Visitor<'de> for OptionalValueVisitor<K, V>
+    where
+        K: Deserialize<'de> + Eq + std::hash::Hash + std::cmp::Ord,
+        V: Deserialize<'de>,
+    {
+        type Value = HashMap<K, Option<V>>;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+            formatter.write_str("HashMap<String, Option<String>>")
+        }
+
+        fn visit_map<M>(self, visitor: M) -> StdResult<Self::Value, M::Error>
+        where
+            M: MapAccess<'de>,
+        {
+            // `MapAccessDeserializer` is a wrapper that turns a `MapAccess`
+            // into a `Deserializer`, allowing it to be used as the input to T's
+            // `Deserialize` implementation. T then deserializes itself using
+            // the entries from the map visitor.
+            Deserialize::deserialize(de::value::MapAccessDeserializer::new(visitor))
+        }
+    }
+
+    let map: StdResult<HashMap<K, Option<V>>, D::Error> =
+        deserializer.deserialize_map(OptionalValueVisitor::new());
+    map.map(|mut hashmap| {
+        hashmap
+            .drain()
```

Comment:
```diff
-.drain()
+.into_iter()
```

(`drain()` would be useful if you wanted to use `hashmap` afterwards, but you don't.)